src/__tests__/linter.test.ts

```diff
@@ -213,7 +214,7 @@ describe('linter', () => {
     expect(result).toEqual([
       expect.objectContaining({
         code: 'oas3-schema',
-        message: 'Property `type` is not expected to be here.',
+        message: '`header-1` property should have required property `schema`.',
```

Comment:
What's going on here? A number of things:
- Originally the `required property` error was getting filtered out in `prepareResults` as a "duplicate" of the `additionalProperties` error since both had the same `basePath`.
- With my change, both errors survive the filtering in `prepareResults`, but then `better-ajv-errors` filters out the `additionalProperties` error because it is a "child" of the `required` error ([ref](https://github.com/stoplightio/better-ajv-errors/blob/master/src/helpers.js#L47)).

I wish `better-ajv-errors` preserved the `additionalProperties` errors, but I think this behavior is arguably no worse than the previous behavior and quite possibly better.

Reply:
> With my change, both errors survive the filtering in prepareResults, but then better-ajv-errors filters out the additionalProperties error because it is a "child" of the required error (ref).

I fixed that! https://github.com/stoplightio/better-ajv-errors/pull/1 - @philsturgeon is on the review there, but do feel free to chime in at your leisure and tell me what you think.
I needed it for the other PR (https://github.com/stoplightio/spectral/pull/1430).